informers.go

```diff
@@ -23,14 +43,122 @@ func getVirtCli() (kubecli.KubevirtClient, error) {
+		DeleteFunc: handleVmiDeletion,
+	})
+
+	return vmiInformer, nil
+}
+
+func updateMachineTypes(virtCli kubecli.KubevirtClient) error {
+	vmList, err := virtCli.VirtualMachine(namespace).List(&k8sv1.ListOptions{})
 	if err != nil {
-		return nil, err
+		return err
+	}
+	for _, vm := range vmList.Items {
+		machineType := vm.Spec.Template.Spec.Domain.Machine.Type
+		machineTypeSubstrings := strings.Split(machineType, "-")
+		version := machineTypeSubstrings[2]
+		if len(machineTypeSubstrings) != 3 {
+			return nil
+		}
+		
+		if strings.Contains(version, "rhel") && version < "rhel9.0.0" {
+			machineTypeSubstrings[2] = latestMachineTypeVersion
+			machineType = strings.Join(machineTypeSubstrings, "-")
+			updateMachineType := fmt.Sprintf(`{"spec": {"template": {"spec": {"domain": {"machine": {"type": "%s"}}}}}}`, machineType)
+			
+			_, err = virtCli.VirtualMachine(vm.Namespace).Patch(vm.Name, types.StrategicMergePatchType, []byte(updateMachineType), &k8sv1.PatchOptions{})
+			if err != nil {
+				return err
+			}
+			
+			// add label to running VMs that a restart is required for change to take place
+			if vm.Status.Ready {
+				// adding the warning label to the VMs regardless if we restart them now or if the user does it manually
+				// shouldn't matter, since the deletion of the VMI will remove the label and remove the vmi list anyway
+				err = addWarningLabel(virtCli, &vm)
+				if err != nil {
+					return err
+				}
+				
+				if restartNow {
+					err = virtCli.VirtualMachine(vm.Namespace).Restart(vm.Name, &k6tv1.RestartOptions{})
+					if err != nil {
+						return err
+					}
+				}
+			}
+		}
 	}
+	return nil
+}
 
-	listWatcher := cache.NewListWatchFromClient(virtCli.RestClient(), "virtualmachineinstances", k8sv1.NamespaceAll, fields.Everything())
-	vmiInformer := cache.NewSharedIndexInformer(listWatcher, &k6tv1.VirtualMachineInstance{}, 1*time.Hour, cache.Indexers{})
+func addWarningLabel (virtCli kubecli.KubevirtClient, vm *k6tv1.VirtualMachine) error {
+	addLabel := fmt.Sprint(`{"metadata": {"labels": {"restart-vm-required": "true"}}}}}`)
+	_, err := virtCli.VirtualMachineInstance(vm.Namespace).Patch(vm.Name, types.StrategicMergePatchType, []byte(addLabel), &k8sv1.PatchOptions{})
+	if err != nil {
+		return err
+	}
+	
+	// get VM name in the format namespace/name
+	vmKey, err := cache.MetaNamespaceKeyFunc(vm)
+	if err != nil {
+		return err
+	}
+	vmisPendingUpdate[vmKey] = struct{}{}
+	
+	return nil
+}
 
-	return vmiInformer, nil
+func handleVmiUpdate(oldObj interface{}, newObj interface{}) {
+	vmi, ok := newObj.(*k6tv1.VirtualMachineInstance)
+	if !ok {
+		return
+	}
+	
+	// check VMI status for Succeeded
+	if vmi.Status.Phase != k6tv1.Succeeded {
+		return
+	}
+	
+	removeWarningLabel(vmi)	
+}
+
+func handleVmiDeletion(obj interface{}) {
+	vmi, ok := obj.(*k6tv1.VirtualMachineInstance)
+	if !ok {
+		return
+	}
+	
+	removeWarningLabel(vmi)
+}
+
+func removeWarningLabel(vmi *k6tv1.VirtualMachineInstance) {
```

Comment:
nit: I'd make the function name a bit more accurate, since it doesn't only remove the label, but also removed the VMI from the map, ends the job if necessary, etc.

WDYT about something like `handleDeletedVmi`?